processing/src/main/java/io/druid/segment/incremental/OnheapIncrementalIndex.java

```diff
@@ -64,14 +73,39 @@
       boolean reportParseExceptions,
       boolean concurrentEventAdd,
       boolean sortFacts,
-      int maxRowCount
+      int maxRowCount,
+      long maxBytesInMemory
   )
   {
     super(incrementalIndexSchema, deserializeComplexMetrics, reportParseExceptions, concurrentEventAdd);
     this.maxRowCount = maxRowCount;
-
+    this.maxBytesInMemory = maxBytesInMemory == 0 ? defaultMaxBytesInMemory : maxBytesInMemory;
     this.facts = incrementalIndexSchema.isRollup() ? new RollupFactsHolder(sortFacts, dimsComparator(), getDimensions())
                                                    : new PlainFactsHolder(sortFacts);
+    maxBytesPerRowForAggregators = getMaxBytesPerRowForAggregators(incrementalIndexSchema);
+
+  }
+
+  /**
+   * Gives estimated max size per aggregator.
+   * size estimation consists of:
+   * <ul>
+   * <li> metrics length : Integer.BYTES * len
+   * <li> maxAggregatorIntermediateSize : getMaxIntermediateSize per aggregator + overhead(16 bytes)
```

Comment:
This deserves a comment about what how we arrived at 16 bytes for "overhead".

Reply:
We arrived at this from https://gist.github.com/arturmkrtchyan/43d6135e8a15798cc46c, but not sure how much to trust this, this page https://www.ibm.com/developerworks/java/library/j-codetoheap/index.html suggests the overhead for object in 64-bit jvm process should be 192 bits instead of 128 bits. Which one should we go with ?